source/gloperate/source/rendering/ScreenAlignedQuad.cpp

```diff
@@ -47,26 +50,28 @@ ScreenAlignedQuad::~ScreenAlignedQuad()
 
 const Drawable * ScreenAlignedQuad::drawable() const
 {
-    return m_drawable;
+    return m_drawable.get();
 }
 
 void ScreenAlignedQuad::draw() const
 {
     m_drawable->draw();
 }
 
-const std::string ScreenAlignedQuad::s_defaultVertexShaderSource = gloperate::dataPath() + "/gloperate/shaders/screenaligned/default.vert";
+const std::unique_ptr<globjects::File> ScreenAlignedQuad::s_defaultVertexShaderSource = globjects::Shader::sourceFromFile(
+            gloperate::dataPath() + "/gloperate/shaders/screenaligned/default.vert");
 
-const std::string ScreenAlignedQuad::s_defaultFragmentShaderSource = gloperate::dataPath() + "/gloperate/shaders/screenaligned/default.frag";
+const std::unique_ptr<globjects::File> ScreenAlignedQuad::s_defaultFragmentShaderSource = globjects::Shader::sourceFromFile(
+            gloperate::dataPath() + "/gloperate/shaders/screenaligned/default.frag");
 
-globjects::Shader* ScreenAlignedQuad::createDefaultVertexShader()
+std::unique_ptr<globjects::Shader> ScreenAlignedQuad::createDefaultVertexShader()
 {
-    return globjects::Shader::fromFile(gl::GL_VERTEX_SHADER, s_defaultVertexShaderSource);
+    return cppassist::make_unique<globjects::Shader>(gl::GL_VERTEX_SHADER, s_defaultVertexShaderSource.get());
 }
 
-globjects::Shader* ScreenAlignedQuad::createDefaultFragmentShader()
+std::unique_ptr<globjects::Shader> ScreenAlignedQuad::createDefaultFragmentShader()
 {
-    return globjects::Shader::fromFile(gl::GL_FRAGMENT_SHADER, s_defaultFragmentShaderSource);
+    return cppassist::make_unique<globjects::Shader>(gl::GL_FRAGMENT_SHADER, s_defaultFragmentShaderSource.get());
```

Comment:
This will not be the best implementation, e.g., for macOS.
As suggested, you should return just the shader source, not a fully instantiated shader.
The caller should be able to integrate shader templates for version fixes or other replacements.